subsys/bluetooth/controller/ticker/ticker.c

```diff
@@ -813,6 +809,10 @@ static uint8_t ticker_resolve_collision(struct ticker_node *nodes,
 			uint8_t next_is_critical = ticker_next->priority ==
 				TICKER_PRIORITY_CRITICAL;
 
+			/* Is the current and next node equal in priority? */
+			uint8_t equal_priority =
+				(ticker->priority == ticker_next->priority);
```

Comment:
why is this using `uint8_t` instead of `bool`?

Reply:
I believe the original contribution comes from where there wasn't `bool` or `_Bool` defined in other compilers/architectures. But that is not the case in the present day. Currently, the controller code is a mix of `uint8_t` and `bool`.

Also, from linux coding style:
> Do not use bool if cache line layout or size of the value matters, as its size
and alignment varies based on the compiled architecture. Structures that are
optimized for alignment and size should not use bool.

> If a structure has many true/false values, consider consolidating them into a
bitfield with 1 bit members, or using an appropriate fixed width type, such as
u8.

I agree `bool` improves readability but at a cost on some architectures.